src/utils.js

```diff
@@ -2,23 +2,35 @@ import redent from 'redent'
 import {
   RECEIVED_COLOR as receivedColor,
   EXPECTED_COLOR as expectedColor,
+  matcherHint,
+  printWithType,
+  printReceived,
 } from 'jest-matcher-utils'
 
-function getDisplayName(subject) {
-  if (subject && subject.constructor) {
-    return subject.constructor.name
-  } else {
-    return typeof subject
+class HtmlElementTypeError extends Error {
+  constructor(received, matcherFn, context) {
+    super()
+
+    /* istanbul ignore next */
+    if (Error.captureStackTrace) {
+      Error.captureStackTrace(this, matcherFn)
+    }
+    this.message = [
+      matcherHint(
+        `${context.isNot ? '.not' : ''}.${matcherFn.name}`,
+        'received',
+        '',
+      ),
+      '',
+      `${receivedColor('received')} value must be an HTMLElement.`,
+      printWithType('Received', received, printReceived),
+    ].join('\n')
```

Comment:
How about using `getMessage` here too? It could read something like this:

```
expect(element).toBeInTheDOM()

Expected:
  <red>element</> must be an HTMLElement
Received:
  object: <red>{\\"thisIsNot\\": \\"an html element\\"}</>"
```

Reply:
This is clone of the `value must be a Promise` error. 'Expected:'  part  is usually used when there are some arguments in the actual matcher e.g. as in `toBeEqual`. Also `printWithType` is not compatible with `getMessage`, but too handy in this case.

Reply:
Ok, fine then.